tests/components/ps4/test_config_flow.py

```diff
@@ -0,0 +1,173 @@
+"""Define tests for the PlayStation 4 config flow."""
+from unittest.mock import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components import ps4
+from homeassistant.components.ps4.const import (
+    DEFAULT_NAME, DEFAULT_REGION)
+from homeassistant.const import (
+    CONF_CODE, CONF_HOST, CONF_IP_ADDRESS, CONF_NAME, CONF_REGION, CONF_TOKEN)
+
+from tests.common import MockConfigEntry, mock_coro
+
+
+MOCK_TITLE = 'PlayStation 4'
+MOCK_CODE = '12345678'
+MOCK_CREDS = '000aa000'
+MOCK_HOST = '192.0.0.0'
+MOCK_DEVICE = {
+    CONF_HOST: MOCK_HOST,
+    CONF_NAME: DEFAULT_NAME,
+    CONF_REGION: DEFAULT_REGION
+}
+MOCK_CONFIG = {
+    CONF_IP_ADDRESS: MOCK_HOST,
+    CONF_NAME: DEFAULT_NAME,
+    CONF_REGION: DEFAULT_REGION,
+    CONF_CODE: MOCK_CODE
+}
+MOCK_DATA = {
+    CONF_TOKEN: MOCK_CREDS,
+    'devices': MOCK_DEVICE
+}
+MOCK_UDP_PORT = int(987)
+MOCK_TCP_PORT = int(997)
+
+
+async def test_full_flow_implementation(hass):
+    """Test registering an implementation and flow works."""
+    flow = ps4.PlayStation4FlowHandler()
+    flow.hass = hass
+
+    # User Step Started, results in Step Creds
+    with patch('pyps4_homeassistant.Helper.port_bind',
+               return_value=mock_coro(return_value=None)):
```

Comment:
The helper method isn't a coroutine. We shouldn't return a coroutine as return value.